src/db.rs

```diff
@@ -89,7 +89,15 @@ pub trait Database: Debug + Clone + Send + Sync {
     async fn state(&self, coordinate: &Coordinate) -> Result<Option<Vec<u8>>, Error>;
 
     /// Set the state of a given [`Coordinate`].
-    async fn set_state(&self, coordinate: &Coordinate, state: &[u8]) -> Result<(), Error>;
+    async fn set_state(
+        &self,
+        coordinate: &Coordinate,
+        state: &[u8],
+        is_branch: &bool,
+    ) -> Result<(), Error>;
+
+    /// Deletes all states with the given repository and is_branch values
+    async fn delete_states(&self, repository: &str, is_branch: bool) -> Result<(), Error>;
```

Comment:
I see we've recorded temporal branch, but what about tags?
As a reminder, this ticket should also temporally track tag state:
> If import_tags goes from true to false, tag state tracked in the DB should be deleted, so that if it is ever flipped to true again this counts as a “first time” import of tags.

Reply:
Currently I have workflow as:

1. Check the config to see what import_tags is set as
2. If import_tags is configured to `false` then delete all references in the db where is_branch is set to `false`
2.a. The idea is that if the reference isn't a branch then it must be a tag

The thought process here was that we could just use the is_branch column in the db to check whether an entry is either a branch or a tag. So whenever import_tags is set to false, we would be deleting all tag entries which can be checked when is_branch is false in the db. I didn't find it necessary to have another column to explicitly check if the entry is a tag (e.g. is_tag) because that could be inferred from is_branch alone. I can see the appeal in having two separate columns to track branch and tag state as it makes our intentions more clear, but would like to hear your thoughts on this as well.

Reply:
That makes sense. Please ensure:
- This is documented under the dev docs in the repo.
- The user-facing implications of this change are documented under the user-facing docs in the repo.